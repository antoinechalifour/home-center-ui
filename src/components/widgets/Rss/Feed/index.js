import { graphql } from 'react-apollo'
import { getRssFeed } from 'queries/rss'
import Feed from './Feed'

export default graphql(getRssFeed)(Feed)
