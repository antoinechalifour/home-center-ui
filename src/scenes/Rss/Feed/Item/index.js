import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { graphql } from 'react-apollo'
import { deleteSource } from 'queries/rss'
import Item from './Item'

class ItemContainer extends Component {
  static propTypes = {
    sourceId: PropTypes.number.isRequired,
    mutate: PropTypes.func.isRequired
  }

  deleteSource = () => {
    this.props.mutate({
      variables: {
        input: { id: this.props.sourceId }
      }
    })
  }

  render () {
    return <Item {...this.props} deleteSource={this.deleteSource} />
  }
}

const queryOptions = {
  options: {
    refetchQueries: ['GetRssQuery']
  }
}

export default graphql(deleteSource, queryOptions)(ItemContainer)
