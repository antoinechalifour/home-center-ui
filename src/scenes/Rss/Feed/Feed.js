import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import Empty from 'ui/Empty'
import Item from './Item'

export default function Feed ({ data }) {
  if (data.loading) {
    return <Empty>Fetching your feed...</Empty>
  } else if (!data.loading && data.feed.length === 0) {
    return (
      <Empty>
        Your feed is empty. Use the "Source" tab to start adding items :)
      </Empty>
    )
  }

  return (
    <Items>
      {data.feed.map(item => <Item key={item.title} {...item} />)}
    </Items>
  )
}

Feed.propTypes = {
  data: PropTypes.shape({
    loading: PropTypes.bool.isRequired,
    feed: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string.isRequired,
        link: PropTypes.string.isRequired,
        date: PropTypes.string.isRequired,
        sourceId: PropTypes.number.isRequired,
        source: PropTypes.string.isRequired,
        content: PropTypes.string
      })
    )
  }).isRequired
}

const Items = styled.ul`
  max-height: 60vh;
  display: flex;
  overflow-y: auto;

  > li {
    flex: 0 0 95%;

    @media (min-width: 860px) {
      flex: 0 0 50%;
    }
  }

  > li + li {
    border-left: 1px solid rgba(0, 0, 0, .05);
  }
`
