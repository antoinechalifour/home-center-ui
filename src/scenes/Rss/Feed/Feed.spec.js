import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import Feed from './Feed'

const renderer = new ShallowRenderer()

describe('scenes/Rss/Feed', () => {
  it('Should render a loader when the data is loading', () => {
    const props = {
      data: { loading: true },
      deleteSource: jest.fn()
    }
    const tree = renderer.render(<Feed {...props} />)

    expect(tree).toMatchSnapshot()
  })

  it('Should render an empty message when the feed is empty', () => {
    const props = {
      data: {
        loading: false,
        feed: []
      },
      deleteSource: jest.fn()
    }
    const tree = renderer.render(<Feed {...props} />)

    expect(tree).toMatchSnapshot()
  })

  it('Should render the feed otherwise', () => {
    const props = {
      data: {
        loading: false,
        feed: [
          {
            date: 'Tue, 26 Dec 2017 09:46:38 +0100',
            link: 'http://fizz.buzz',
            source: 'FizzBuzz Magazine',
            title: 'Fizz! Buzz!'
          },
          {
            date: 'Tue, 26 Dec 2017 09:42:38 +0100',
            link: 'http://foo.bar',
            source: 'FooBar Magazine',
            title: 'Foo! Bar!'
          }
        ]
      },
      deleteSource: jest.fn()
    }
    const tree = renderer.render(<Feed {...props} />)

    expect(tree).toMatchSnapshot()
  })
})
