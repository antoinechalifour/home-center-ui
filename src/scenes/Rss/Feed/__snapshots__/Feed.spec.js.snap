// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scenes/Rss/Feed Should render a loader when the data is loading 1`] = `
<styled.div>
  Fetching your feed...
</styled.div>
`;

exports[`scenes/Rss/Feed Should render an empty message when the feed is empty 1`] = `
<styled.div>
  Your feed is empty. Use the "Source" tab to start adding items :)
</styled.div>
`;

exports[`scenes/Rss/Feed Should render the feed otherwise 1`] = `
<styled.ul>
  <Apollo(ItemContainer)
    date="Tue, 26 Dec 2017 09:46:38 +0100"
    link="http://fizz.buzz"
    source="FizzBuzz Magazine"
    title="Fizz! Buzz!"
  />
  <Apollo(ItemContainer)
    date="Tue, 26 Dec 2017 09:42:38 +0100"
    link="http://foo.bar"
    source="FooBar Magazine"
    title="Foo! Bar!"
  />
</styled.ul>
`;
